src/connectors/voice-search/__tests__/connectVoiceSearch-test.js

```diff
@@ -0,0 +1,188 @@
+    widget._voiceSearchHelper.changeState();
+    expect(renderFn).toHaveBeenCalledTimes(2);
+    widget._voiceSearchHelper.changeState();
+    expect(renderFn).toHaveBeenCalledTimes(3);
+  });
+
+  it('setQuery and search when query changes', () => {
+    const { widget, helper } = getDefaultSetup();
+    jest.spyOn(helper, 'setQuery');
+    helper.search = jest.fn();
+    widget.init({ helper });
+    widget._voiceSearchHelper.changeQuery('foo');
+    expect(helper.setQuery).toHaveBeenCalledTimes(1);
+    expect(helper.setQuery).toHaveBeenCalledWith('foo');
+    expect(helper.search).toHaveBeenCalledTimes(1);
+  });
+
+  describe('getWidgetState', () => {
+    it('returns the same state if query is an empty string', () => {
+      const { widget, helper, refine } = getInitializedWidget();
+      const uiStateBefore = { foo: 'bar' };
+      refine('');
+      const uiStateAfter = widget.getWidgetState(uiStateBefore, {
+        searchParameters: helper.state,
+      });
+      expect(uiStateAfter).toBe(uiStateBefore);
+    });
+
+    it('returns the same state if query is same', () => {
+      const { widget, helper, refine } = getInitializedWidget();
+      refine('myQuery');
+      const uiStateBefore = widget.getWidgetState(
+        {},
+        {
+          searchParameters: helper.state,
+        }
+      );
+      const uiStateAfter = widget.getWidgetState(uiStateBefore, {
+        searchParameters: helper.state,
+      });
+      expect(uiStateAfter).toBe(uiStateBefore);
+    });
+
+    it('returns new state with query after refine', () => {
+      const { widget, helper, refine } = getInitializedWidget();
+      const uiStateBefore = { foo: 'bar' };
+      refine('myQuery');
+      const uiStateAfter = widget.getWidgetState(uiStateBefore, {
+        searchParameters: helper.state,
+      });
+      expect(uiStateAfter).toEqual({
+        foo: 'bar',
+        query: 'myQuery',
+      });
+    });
+  });
+
+  describe('getWidgetSearchParameters', () => {
+    it('should return the same SearchParameters if no value is in the UI state', () => {
+      const { widget, helper } = getInitializedWidget();
+      const uiState = {};
+      const searchParametersBefore = SearchParameters.make(helper.state);
+      const searchParametersAfter = widget.getWidgetSearchParameters(
+        searchParametersBefore,
+        { uiState }
+      );
+      expect(searchParametersAfter).toBe(searchParametersBefore);
+    });
+
+    it('should add the refinement according to the UI state provided', () => {
+      const { widget, helper } = getInitializedWidget();
+      const uiState = {
+        query: 'my search',
+      };
+      const searchParametersBefore = SearchParameters.make(helper.state);
+      const searchParametersAfter = widget.getWidgetSearchParameters(
+        searchParametersBefore,
+        { uiState }
+      );
+      expect(searchParametersAfter.query).toEqual(uiState.query);
+      expect(searchParametersAfter).toMatchSnapshot();
```

Comment:
What do you expect to catch with this snapshot? Is it necessary?

Reply:
You're right. It's not necessary and doesn't give any benefit. I've removed them.